src/Integration.Vsix/SonarLintTagger/VsAwareAnalysisService.cs

```diff
@@ -29,11 +30,13 @@ namespace SonarLint.VisualStudio.Integration.Vsix;
 
 internal interface IVsAwareAnalysisService
 {
-    void RequestAnalysis(string filePath, 
-        ITextDocument document,
+    void RequestAnalysis(ITextDocument document,
```

Comment:
The number of parameters keep increasing unnecesarily. Please use the AnalysisSnapshot data structure mentioned in the comment from TextBufferIssueTracker